src/test/java/ro/developmentfactory/arrayProblem/MainTest.java

```diff
@@ -0,0 +1,109 @@
+package ro.developmentfactory.arrayProblem;
+
+import static org.junit.jupiter.api.Assertions.*;
+import static org.mockito.Mockito.when;
+import org.junit.jupiter.api.AfterEach;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.mockito.Mock;
+import org.mockito.MockitoAnnotations;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+import java.util.concurrent.ExecutorService;
+import java.util.concurrent.Executors;
+import java.util.concurrent.ThreadPoolExecutor;
+import java.util.concurrent.TimeUnit;
+
+
+class MainTest{
+
+    @Mock
+    IntermediateSuma mockIntermediateSuma;
+
+    @BeforeEach
+    void setUp(){
+        MockitoAnnotations.initMocks(this);
+    }
+
+    @AfterEach
+    void tearDown(){
+    }
+
+    @Test
+    void testExecutorServiceCreation(){
+        ExecutorService executor = Executors.newFixedThreadPool(10);
+        assertNotNull(executor);
+        assertTrue(executor instanceof ThreadPoolExecutor);
+        assertEquals(10, ((ThreadPoolExecutor) executor).getCorePoolSize());
+        executor.shutdown();
+    }
+
+    @Test
+    void testArrayChunking(){
```

Comment:
we should not care about the "internals" of your implementation to this level. even if the task was:

> Write a program that will accept a large array of integers as input and calculate the sum of all integers in 5 threads, then add the result together, returning the sum.

it's quite hard to check this without assuming how it works under the hood (`array.length / numberOfTasks`) and in tests we should never assume that we know these details. in this case, we are more interested that by the end the right sum is calculated using your interfaces. you might want to do an extra step to have your code more testable: either write `CinemaService` that will accept either `OptimisticLocking` or `PessimisticLocking` in its constructor, `CinemaService` will have then only 1 method: `bookSeats(int numSeats)`; or write `CinemaService` with 2 methods: `bookSeatsOptimistic(int numSeats)` and `bookSeatspessimistic(int numSeats)` and `CinemaService` should accept both `OptimisticLocking` and `PessimisticLocking` in its constructor. the 1st option will produce a more clean code in this case